### DISCUSSION TITLE
Error: Unable to retrieve the document from the global scope.

### DISCUSSION THREAD

**Discussion by Yokubjon-J** [remirror/remirror#2095] [Q&A]
I was implementing SSR with react. For this purpose I have installed emotion/server, emotion/cache and express (and webpack!). But I have encountered a problem with rendering remirror on the screen. I believe the problem is in my server.js file, which is an entry file for webpack. Due `ReactDOMServer.renderToString` in [this line](https://github.com/Yokubjon-J/exp-webpack-css-error-mrr/blob/master/server.js#L18), my program is throwing:

```
Error: Unable to retrieve the document from the global scope. 
It seems that you are running Remirror in a non-browser environment. Remirror need browser APIs to work. 
If you are using Jest (or other testing frameworks), make sure that you are using the JSDOM environment (https://jestjs.io/docs/29.0/configuration#testenvironment-string). 
If you are using Next.js (or other server-side rendering frameworks), please use dynamic import with `ssr: false` to load the editor component without rendering it on the server (https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr). 
If you are using Node.js, you can install JSDOM and Remirror will try to use it automatically, or you can create a fake document and pass it to Remirror
    at getDocument (C:\Users\User\desktop\projects\mini-repo-css-bundling\node_modules\@remirror\core-utils\dist\remirror-core-utils.cjs:1148:9)
    at _RemirrorManager.get document [as document] (C:\Users
```
However, If I just remove the `ReactDOMServer.renderToString` part from that line, I get the error `Hydration failed because the initial UI does not match what was rendered on the server.`

I run my program by typing `npm run development`.

I want to know your thoughts on the problem. Please let me know if I missed some detail.

**Comment by armannaj** [comment]
I'm facing the same error and issue. However, it doesn't block rendering the page, and the page fully works fine.

This error is preventing search engines to index the page though
